src/ripple/app/ledger/LedgerTiming.cpp

```diff
@@ -61,119 +56,117 @@ bool ContinuousLedgerTiming::shouldClose (
             return true;
         }
 
-#if 0 // This false triggers on the genesis ledger
-
-        if (previousMSeconds > (1000 * (LEDGER_IDLE_INTERVAL + 2))) // the last ledger was very slow to close
-        {
-            WriteLog (lsTRACE, LedgerTiming) << "was slow to converge (p=" << (previousMSeconds) << ")";
-
-            if (previousMSeconds < 2000)
-                return previousMSeconds;
-
-            return previousMSeconds - 1000;
-        }
 
-#endif
+        // Only close if we have idled for too long.
         return currentMSeconds >= (idleInterval * 1000); // normal idle
     }
 
-    if ((openMSeconds < LEDGER_MIN_CLOSE) && ((proposersClosed + proposersValidated) < (previousProposers / 2 )))
+    // If we have any transactions, we don't want to close too frequently:
```

Comment:
Am I missing something?  What's the advantage of splitting this comparison into two?